src/tensorflow.cc

```diff
@@ -1411,66 +1621,33 @@ AutoCompleteHelper::FixIOConfig(
-      // The number of elements in dims should match 'rank - 1'
-      // when the model supports batching; otherwise, number of
-      // elements in dims should match 'rank'. This does not
-      // try to overwrite the user provided configuration, throws
-      // error instead.
-      // However, ragged batching is an exception to this rule. A
-      // tensor allowing ragged batch should not match with
-      // 'rank - 1'.
-      size_t io_size = ios.ArraySize();
-      for (size_t i = 0; i < io_size; ++i) {
-        triton::common::TritonJson::Value current_io_object(
-            model_state_->ModelConfig(),
-            triton::common::TritonJson::ValueType::OBJECT);
-        ios.IndexAsObject(i, &current_io_object);
-
-        if (!using_ragged_batching_) {
-          triton::common::TritonJson::Value current_dims(
-              model_state_->ModelConfig(),
-              triton::common::TritonJson::ValueType::ARRAY);
-          current_io_object.Find("dims", &current_dims);
-
-          if (model_support_batching_) {
-            RETURN_ERROR_IF_TRUE(
-                current_dims.ArraySize() != (io->shape_->rank_ - 1),
-                TRITONSERVER_ERROR_INVALID_ARG,
-                std::string(
-                    "Number of dimensions (" +
-                    std::to_string(current_dims.ArraySize()) +
-                    ") given for tensor " + io->name_ + " for model '" +
-                    model_state_->Name() +
-                    "' in configuration does not match the rank (" +
-                    std::to_string(io->shape_->rank_ - 1) +
-                    ") of the loaded model."));
-          } else {
-            RETURN_ERROR_IF_TRUE(
-                current_dims.ArraySize() != io->shape_->rank_,
-                TRITONSERVER_ERROR_INVALID_ARG,
-                std::string(
-                    "Number of dimensions (" +
-                    std::to_string(dims.ArraySize()) + ") given for tensor " +
-                    io->name_ + " for model '" + model_state_->Name() +
-                    "' in configuration does not match the rank (" +
-                    std::to_string(io->shape_->rank_) +
-                    ") of the loaded model."));
-          }
-        }
-      }
+    }
+    if (found_ios) {
+      ios.Swap(auto_complete_ios);
     } else {
+      model_state_->ModelConfig().Add("output", std::move(auto_complete_ios));
+    }
+  } else {
+    // don't need to check what was processed at the end so no need to copy
+    triton::common::TritonJson::Value current_io(
+        model_state_->ModelConfig(),
+        triton::common::TritonJson::ValueType::OBJECT);
+    for (size_t i = 0; i < ios.ArraySize(); ++i) {
+      ios.IndexAsObject(i, &current_io);
+
+      std::string output_name;
+      RETURN_IF_ERROR(current_io.MemberAsString("name", &output_name));
+
+      TRITONTF_IO* io = FindModelIOByName(output_name.c_str(), reference_list);
       RETURN_ERROR_IF_TRUE(
```

Comment:
If not found then just skip this output. ValidateTRITONTFModel will take care of the validation.

Reply:
Confirmed model validation takes care of this